src/v/serde/test/serde_test.cc

```diff
@@ -815,4 +815,14 @@ SEASTAR_THREAD_TEST_CASE(absl_node_hash_set) {
     set_test<absl::node_hash_set>();
 }
 
-SEASTAR_THREAD_TEST_CASE(absl_btree_set) { set_test<absl::btree_set>(); }
\ No newline at end of file
+SEASTAR_THREAD_TEST_CASE(absl_btree_set) { set_test<absl::btree_set>(); }
+
+struct foo_bar {};
+
+SEASTAR_THREAD_TEST_CASE(type_str) {
+    BOOST_CHECK_EQUAL(serde::type_str<int>(), "int");
+    BOOST_CHECK_EQUAL(serde::type_str<void>(), "void");
+    BOOST_CHECK_EQUAL(
+      serde::type_str<std::string>(), "std::basic_string<char>");
```

Comment:
i am going to drop this test. as it fails on arm64:
```
  | ../../../src/v/serde/test/serde_test.cc(826): error: in "type_str": check serde::type_str<std::string>() == "std::basic_string<char>" has failed [std::string != std::basic_string<char>]
```

i failed to reproduce this issue with clang-12 + libstdc++11. but it turns out the class name of specialized C++ template class in STL is subject to change. we could have something like `std::__cxx11::basic_string<char>` with GCC 12.1.

see https://godbolt.org/z/YMEd1ahz8